test/karma/directives/routeOnClick.test.js

```diff
@@ -22,10 +22,68 @@ describe('routeOnClick directive', function() {
       });
     });
   });
+
+  describe('when element is an Anchor tag', function() {
+    it('should populate the href attribute of the element with the URL returned from the matching URL writer', function () {
+      window.angular.mock.module(function (RouteProvider) {
+        RouteProvider.registerUrlWriter('routeLink', () => '/contacts/update/1');
+        RouteProvider.setHtml5Mode(false);
+      });
+
+      inject(function ($rootScope, $compile) {
+        let element = $compile('<a route-on-click="routeLinkUrlWriter()">Link</a>')($rootScope.$new());
+        $rootScope.$digest();
+        expect(element.attr('href')).toBe('#/contacts/update/1');
+      });
+    });
+
+    it('should update the href attribute when the expression eval changes', function() {
+      window.angular.mock.module(function(RouteProvider) {
+        RouteProvider.setPersistentStates('page');
```

Comment:
Do you need RouteProvider.setPersistentStates('page') ?

Reply:
yup it needs it blows up otherwise